src/chain_monitor.ts

```diff
@@ -0,0 +1,123 @@
+import { ExtendedAccount } from "./config"
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { Hash, Header } from "@polkadot/types/interfaces/runtime";
+import { GenericExtrinsic } from "@polkadot/types/"
+
+import "@polkadot/api-augment";
+import "@polkadot/types-augment";
+
+const DEBUG = false;
+
+export interface ExtrinsicItem {
+    index: number,
+    section: string,
+    method: string,
+    account: ExtendedAccount,
+    data: string
+}
+
+export interface EventItem {
+    section: string,
+    method: string,
+    data: string,
+}
+
+// This holds the raw blockdata,
+// the blockhandler extracts relevant data into a "Report" item.
+export interface blockData {
+    chain: string,
+    number: number,
+    hash: Hash,
+    timestamp: number,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    events: any[], //any is correct for now because it's a mutable type
+    extrinsics: GenericExtrinsic[]
+}
+
+export interface Report {
+    chain: string,
+    blocknumber: number,
+    hash: Hash,
+    timestamp: number,
+    extrinsics?: ExtrinsicItem[],
+    events?: EventItem[],
+}
+
+
+export class ChainMonitor {
+    private provider!: WsProvider;
+    private api!: ApiPromise;
+    chain!: string;
+
+
+    constructor(
```

Comment:
I don't know what formetter you use here (or none), but I don't agree with it :D